GSoC contributor blog for Security Scan Probe (#6539)
jenkins-infra/jenkins.io: content/blog/2023/07/16/2023-07-16-security-scan-probe-blog.adoc

---
layout: post
title: Security Scan Probe
tags:
- gsoc
- gsoc2023
- healthscore
- probes
- plugin
authors:
- jagruti
opengraph:
  image: /images/gsoc/2023/jagruti/blog_20230716-ssp.png
sig:
- gsoc
---

> == Introduction
> The Jenkins Security team provides a GitHub action to automatically scan the plugin source code during Continuous Integration, and report security issues.
> 
> It is considered a good practice by the Jenkins Community to use such tools to ensure a good security level of plugins.

Reviewer: ```suggestion
== Introduction

The Jenkins Security team offers a GitHub action that automatically scans the plugin source code during Continuous Integration and reports security issues.

Using such tools to ensure a high-security level for plugins is considered a good practice by the Jenkins Community.
```

> Security Scan probe identifies whether the plugin is configured to execute the security action.

Reviewer: ```suggestion
The Security Scan probe identifies whether the plugin is configured to execute the security action.
```

> == Importance of the probe
> This probe is important from the security perspective. As it looks for known security breaches and vulnerability in a plugin.
> 
> This probe ensures that the security scan is configured in the plugin's GitHub Action.

Reviewer: ```suggestion
== Importance of the Probe

This probe is essential from a security perspective as it checks for known security breaches and vulnerabilities in a plugin.

Its primary purpose is to verify that the security scan is properly configured in the plugin's GitHub Action.
```

> == Challenges
> An existing contributor initiated this probe . For some reason they couldn't continue so I took over their work.
> 
> This probe also involved refactoring existing classes to avoid code duplication. As JEP-229 (Continuous  Delivery) probe looks for CI configurations in GitHub Actions as well.

Reviewer: ```suggestion
== Challenges

This probe was initially initiated by an existing contributor, but for some reason, they couldn't continue, and I took over their work.

Additionally, this probe required refactoring existing classes to eliminate code duplication, as the JEP-229 (Continuous Delivery) probe also looks for CI configurations in GitHub Actions.
```

> == Outcome
> This probe will be beneficial to the security team in identifying gaps and improve the security for Jenkins plugin ecosystem.

Reviewer: ```suggestion
== Outcome

This probe will be beneficial to the security team as it helps identify gaps and improve the security of the Jenkins plugin ecosystem.
```

> == Learning
> Rebased an existing Git branch.
> Wrote unit test cases.
> 
> Learned how to benefit from class hierarchy. Built an abstract probe that can be implemented by any probes that want to check their GitHub Action configuration.

Reviewer: ```suggestion
== Learning

During this project, I accomplished the following:
- Successfully rebased an existing Git branch.
- Developed and wrote unit test cases.
- Gained knowledge on leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.
```

> == Links
> For more information or to find answers to any questions you might have, head to the official GSoC 2023 project link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to  "Plugin Health Score"] description page.

Reviewer: ```suggestion
== Links

For more information or if you have any questions, please visit the official GSoC 2023 project link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to  "Plugin Health Score"]) description page.
```

---
layout: post
title: Security Scan Probe
tags:
- gsoc
- gsoc2023
- healthscore
- probes
- plugin
authors:
- jagruti
opengraph:
  image: /images/gsoc/2023/jagruti/blog_20230716-ssp.png
sig:
- gsoc
---

== Introduction

The Jenkins Security team offers a GitHub action that automatically scans the plugin source code during Continuous Integration and reports security issues.

> Using such tools to ensure a high-security level for plugins is considered a good practice by the Jenkins Community.

Reviewer 2: ```suggestion
The Jenkins community considers using such tools to ensure a high-security level for plugins as a good practice.
```
suggestion for wording/sentence structure

The Security Scan probe identifies whether the plugin is configured to execute the security action.

== Importance of the Probe

This probe is essential from a security perspective as it checks for known security breaches and vulnerabilities in a plugin.

Its primary purpose is to verify that the security scan is properly configured in the plugin's GitHub Action.

== Challenges

> This probe was initially initiated by an existing contributor, but for some reason, they couldn't continue, and I took over their work.

Reviewer 2: ```suggestion
This probe was originally started by another contributor.
However, they were unable to continue, so I took over their work.
```
suggestion for wording/sentence structure

> Additionally, this probe required refactoring existing classes to eliminate code duplication, as the JEP-229 (Continuous Delivery) probe also looks for CI configurations in GitHub Actions.

Reviewer 2: ```suggestion
Additionally, this probe required refactoring existing classes to eliminate code duplication, as the link:https://github.com/jenkinsci/jep/tree/master/jep/229[JEP-229 (Continuous Delivery)] probe also looks for CI configurations in GitHub Actions.
```
suggestion to add link for JEP-229

== Outcome

This probe will be beneficial to the security team as it helps identify gaps and improve the security of the Jenkins plugin ecosystem.

== Learning

> During this project, I accomplished the following:

Reviewer 3: ```suggestion
During this project, I accomplished the following:

```

- Successfully rebased an existing Git branch.
- Developed and wrote unit test cases.

> - Gained knowledge on leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.

Reviewer 3: The unordered list here somehow isn't rendering properly, can we fix this?
